File: yh_check/scripts/yh_check_sub.py
#!/usr/bin/python3
#-*- coding: utf-8 -*-

import rospy
from yh_check.msg import YhCheck

class MyCheck:
    def __init__(self):
        self.sub_distance = rospy.Subscriber("check_distance", YhCheck, self.distanceCallback)
        self.sub_camera = rospy.Subscriber("check_camera", YhCheck, self.cameraCallback)
        self.distance = True
        self.camera = True

    def distanceCallback(self, msg): #클래스 안에서 생성하려면 무조건 self를 작성해야함.
        self.distance = msg.data
        if self.distance and self.camera:
            rospy.loginfo("OK")

    def cameraCallback(self, msg):
        self.camera = msg.data
        if self.distance and self.camera:
            rospy.loginfo("OK")

if __name__ == "__main__":
    rospy.init_node("yh_check_sub")
    my_check = MyCheck()
    rospy.spin()



'''class Calculator:
    def __init__(self): # 첫번째는 무조건 self
        self.result = 0

    def add(self, num):
        self.result += num
        return self.result

    def sub(self, num):
        self.result -= num
        return self.result

    def mul(self, num):
        self.result *= num
        return self.result

    def div(self, num):
        return self.result / num  # 바로계산을해버림

cal = Calculator()
cal.add(10) 
cal.sub(4)
cal.mul(2)
cal.div(3)

print(cal.add(4))
print(cal.add(1))
print(cal.add(5))
print(cal.add(8))
print(cal.add(-14))
print(cal.add(40))

class MyCalculator(Calculator):
'''

File: yh_check/src/yh_check_sub.cpp
#include "ros/ros.h"
#include "yh_check/YhCheck.h"

/*class MySub
{
    public:
        //함수들
        MySub(void) = init
        {
            sub_distance = nh.subscribe("check_distance", 10, &MySub::distanceCallback, this);
            sub_camera = nh.subscribe("check_camera", 10, &MySub::cameraCallback, this);
        }
    private:
        //변수들
        ros::NodeHandle nh;
        ros::Subscriber sub_distance;
        ros::Subscriber sub_camera;
        bool distance = true;
        bool camera = true;
};  c++ 클래스는 중괄호 끝에 반드시 세미콜론을 붙여야함
*/

bool distance = true; // 전역변수 선언, 여러곳에서 동시에 쓰겠다.
bool camera = true;

void distanceCallback(const yh_check::YhCheck::ConstPtr& msg)
{
    distance = msg->data;
    if (distance && camera)
    {
        ROS_INFO("OK");
    }
}

void cameraCallback(const yh_check::YhCheck::ConstPtr& msg)
{
    camera = msg->data;
    if (distance && camera)
    {
        ROS_INFO("OK");
    }
}

int main(int argc, char** argv)
{
    ros::init(argc, argv, "yh_check_sub");
    ros::NodeHandle nh;

    ros::Subscriber sub_distance = nh.subscribe("check_distance", 10, distanceCallback);
    ros::Subscriber sub_camera = nh.subscribe("check_camera", 10, cameraCallback);

    ros::spin();

    return 0;
}